pkg/components/cert-manager/component.go

```diff
@@ -95,8 +96,13 @@ func (c *component) RenderManifests() (map[string]string, error) {
 
 func (c *component) Metadata() components.Metadata {
 	return components.Metadata{
-		Name:      name,
-		Namespace: c.Namespace,
+		Name: name,
+		ReleaseNamespace: k8sutil.Namespace{
+			Name: c.Namespace,
+			Labels: map[string]string{
+				"certmanager.k8s.io/disable-validation": "true",
```

Comment:
I think we can have these labels populated dynamically without having to hard code it in the go code. This can be populated when we parse the namespace object and skip from installing it. Like it is done in e2160428cc1eba8e74817efe6db4fd570f58de96.

The only change we need to make is that make metadata object persistent somehow. And that can be done by storing the metadata object in the component object like 391ccbb0ed41968bc6c787e39eb842805841819b.

Also we need to make sure we return pointer to metadata and not metadata object itself. Like 2ba87e7b2a72c418ed06b9fecd195adfaf6b2ee9.

Reply:
I think I stepped on your toes here Imran by creating implementation of what is done in this PR. I was trying to find out what is the best way to provide labels and annotations for the linkerd PR and went down this rabbit hole.

Reply:
How do I proceed with this then ? after many iterations this PR is close to getting merged.

Reply:
I think they can be added later as well. Go ahead with this PR and then I can make necessary fixes in my own PR.